### DISCUSSION TITLE
Estimating Convolution Kernel from Input and Output Images

### DISCUSSION THREAD

**Discussion by liyunfei5126** [RoyiAvital/StackExchangeCodes#7] [Q&A]
Hi Royi,
I was searching for an answer to a similar question that had bothered me for a long time and I came across you answer to the question at https://dsp.stackexchange.com/questions/61043/estimating-convolution-kernel-from-input-and-output-images . I think this is exactly the solution I am looking for. However, due to my very limited knowledge in math, it is difficult for me to really understand the concept, especially how you arrive at the fourth equation. I could not read Matlab code. Is it possible that I can reach you via email so that I can share with your my research question and get some advice from you?
Thanks and best regards,
Yunfei Li

**Comment by RoyiAvital** [comment]
Hi,

You may ask here, I'd try to answer.
But the best would be opening a question on DSP StackExchnage and put the link here, I will try to answer it.

**Reply by liyunfei5126** [reply]
Thanks a lot for the quick reply. Unfortunately I am not yet able to aks question there due to lack of reputation.
Actually, this is a fundamental technical question of one of my research work, if you are interested, I could share it with you privately. I want to do the similar thing as in that question.

Basically, I have to matrice, the original one $A$ and the transformed one $B$. The assumption is that $B$ is the result of the convolution of $A$ and a kernel $K$ with known size, plus a bit of noise. I want to approximate the kernel $K$. I am getting lost in your answer about how you arrive at $X^T (Xh -y)$ and the equation further after it. I tried to build an equation set following your answer but get lost soon. This probably is a very stupid question and I apologize first if this annoys you.
Best regards,
Yunfei Li

**Reply by RoyiAvital** [reply]
Hi,

You don't need reputation to **open new question**.
What programming language are you using?

**Reply by liyunfei5126** [reply]
As it is almost the same question, opening a new one would be against the principle of the site. I will give it a try anyway.

I mainly use R, and sometimes use python and C++ when necessary. 

Thanks a lot and best.

**Reply by liyunfei5126** [reply]
I posted a new question at https://dsp.stackexchange.com/q/84301/64191 .
Please have a look if you are interested.
Thanks in advance and best regards,
Yunfei
